src/extension/vscode/server.ts

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { WebSocketServer, RawData, WebSocket } from "ws";
+import type { MessageAdapter } from "../messageAdapters";
+import { createRpcClient, createRpcHandler } from "../rpc";
+import type { Actor } from "../actor";
+import { createActor } from "../actor";
+import type { ApolloClientInfo } from "../../types";
+import { createId } from "../../utils/createId";
+import allErrorCodes from "../../../all-clients/errorcodes.json";
+import { restoreErrorCodes } from "../../../all-clients/restore-errorcodes.mjs";
+
+// copied into this file to ease type bundling for now
+export interface PublicMessageAdapter {
+  addListener: (listener: (message: unknown) => void) => () => void;
+  postMessage: (message: unknown) => void;
+}
+
+export function runServer(
+  server: WebSocketServer,
+  adapter: PublicMessageAdapter
+) {
+  const clients = new Map<
+    string,
+    {
+      rpcClient: ReturnType<typeof createRpcClient>;
+      actor: Actor;
+      info: ApolloClientInfo;
+    }
+  >();
+
+  const messageAdapter = adapter as MessageAdapter;
+  const tab = createActor(messageAdapter);
```

Comment:
[nit] No need to call this `tab` since we're in a different context (unless you prefer this). I think the browser extension calls it this since it represents the code running in the user's browser tab, but that might not make much sense here.